bin/tx-prover/src/commands/mod.rs

```diff
@@ -96,10 +100,19 @@ impl ProxyConfig {
             .write(config_as_toml_string.as_bytes())
             .map_err(|err| format!("error writing to file: {err}"))?;
 
-        println!("Config updated successfully");
+        debug!("Config updated successfully");
 
         Ok(())
     }
+
+    /// Updates the workers in the configuration with the new list.
+    pub(crate) fn set_workers(workers: Vec<WorkerConfig>) -> Result<(), String> {
```

Comment:
Maybe this could be done as part of the incoming follow-up work, but we should probably check (unless it's being dome somewhere already) that there are no duplicate workers at any point (both in the worker list and persisted config file) to avoid problems if the user accidentally adds the same address/port twice.

Reply:
It is not being checked. Though the worker config is planned to be removed from the configuration file. I think we can dismiss this for now and use that issue to fix this.